iavl/iavl_node.go

```diff
@@ -522,5 +552,31 @@ func removeOrphan(t *IAVLTree, node *IAVLNode) {
 	if t.ndb == nil {
 		return
 	}
+
+	// Don't orphan persistent nodes if another tree depends on it
+	if Committed != nil && t != Committed {
```

Comment:
I wasn't happy with it, but the code underneath needs to know about the various trees that get created and used above. I tried linking the trees together in Copy as a list but that seemed more fragile.